### DISCUSSION TITLE
Can't use pino with transport in production (transport are tree shaken away from build)

### DISCUSSION THREAD

**Discussion by leo91000** [nitrojs/nitro#2011] [Q&A]
Hey guys, 

I want to use pino in my nitro app, however when building the app, only the used modules are packaged to the `.output` dir.

However, pino transport are declared by a string :

```ts
createTransport({ target: 'pino-pretty' }) // target is the name of the transport package
```

So you can see why this won't work with nitro : when running pnpm run build, the package `pino-pretty` is absent, and the server can't start (it crashes).

Do you see a config/workaround to make nuxt include this package to the `.output/server/node_modules/` folder ?

Thanks by advance

**Comment by pi0** [comment]
Hi. Can you please share a runnable repo or reproduction? 🙏🏼

**Reply by leo91000** [reply]
Hi !
I tried to make a reproduction, but it seems to work on the minimal reproduction setup  https://stackblitz.com/edit/github-a4yom5?file=routes%2Findex.ts,package.json 

Could it be because I'm on a monorepo setup, and the logging thing is on it's own package?

Otherwise I found a workaround, with pino stream, you can directly import the transport (however, not ideal performance since  it does not allow for logs to be handled in a worker thread)

```
import pino from 'pino'
import pinoPretty from 'pinoPretty'

const streams = [
  { stream: pino.destination(1) },
  pinoPretty({ ... })
]

const logger = pino(
  {
    level: 'trace',
  },
  pino.multistream(streams)
)
```
